include/zephyr/net/wifi.h

```diff
@@ -536,6 +536,40 @@ static inline const char *get_ps_config_err_code_str(int16_t err_no)
 	return "<unknown>";
 }
 
+/** Promiscuous mode operational settings */
+enum wifi_promiscuous_mode {
+	/** Disable promiscuous Mode setting */
+	WIFI_PROMISC_DISABLE = 0,
+	/** Monitor sniffer mode setting enable */
+	WIFI_PROMISC_MONITOR_MODE,
+	/** STA sniffer mode setting enable */
+	WIFI_PROMISC_STA_MODE,
+	/** AP sniffer mode setting enable */
+	WIFI_PROMISC_AP_MODE,
+	/** STA-AP sniffer mode setting enable */
+	WIFI_PROMISC_STA_AP_MODE,
+};
+
+/** Promiscuous mode filter settings */
+enum wifi_promiscuous_filter {
+	/** Support mgmt, data and ctrl packet sniffing */
+	WIFI_PROMISC_ALL = BIT(0),
```

Comment:
It's not clear to me, why for the mode we use values of range 0-4, but for filter we use bitfields? Can filter enums be combined, so for example we could filter DATA and CTRL packets by setting filter to `(WIFI_PROMISC_DATA | WIFI_PROMISC_CTRL)`?

Reply:
Hi Robert,

Yes, filter settings can be combined. we can enable to set the lower layer to support sniffing of data + ctrl as specified by you. Hence, it was set to be a bit field. 
For the mode of operation, two modes of operation cannot be combined. hence, the use of separate enum values.

Reply:
Thanks for clarifying this. A follow up question then - wouldn't it make sense then for `WIFI_PROMISC_ALL` to be a bitwise sum of all of the filter types (`WIFI_PROMISC_MGMT | WIFI_PROMISC_DATA | WIFI_PROMISC_CTRL`)? In theory, it should have the same effect, so I don't see a reason to differentiate here.